src/components/Expander/ExpanderGroup.tsx

```diff
@@ -82,43 +99,72 @@ const OpenAllButton = ({
   onClick,
   OpenAll,
   CloseAll,
+  onClickAll,
 }: OpenAllButtonProps) => {
   return !!allOpen
-    ? openAllButtonProps(onClick)(CloseAll)
-    : openAllButtonProps(onClick)(OpenAll);
+    ? openAllButtonProps(onClick, onClickAll)(CloseAll)
+    : openAllButtonProps(onClick, onClickAll)(OpenAll);
+};
+
+const InitialStateOfExpanders = (
+  children: Array<React.ReactElement<ExpanderProps>>,
+) => {
+  const openExpanders: number[] = [];
+  React.Children.forEach(children, (child, index) => {
+    if (React.isValidElement(child)) {
+      if (child.props.defaultOpen || child.props.open) {
+        openExpanders.push(index);
+      }
+    }
+  });
+  return openExpanders;
 };
 
 export class ExpanderGroup extends Component<ExpanderGroupProps> {
   state: ExpanderGroupState = {
-    openExpanders: [],
+    openExpanders: InitialStateOfExpanders(this.props.children),
+    toggleAllExpanderState: {
+      toState:
+        InitialStateOfExpanders(this.props.children).length ===
+          this.props.children.length && this.props.children.length > 0,
+    },
   };
 
   handleClick = (index: number = 0) => {
-    const { openExpanders: prevOpenExpanders } = this.state;
-    const prevExpanderOpen = prevOpenExpanders.includes(index);
-    const openExpanders = prevExpanderOpen
-      ? prevOpenExpanders.filter((value) => value !== index)
-      : Array.from(new Set([...prevOpenExpanders, index]));
-    this.setState({ openExpanders });
+    this.setState((prevState: ExpanderGroupState) => {
+      const { openExpanders: prevOpenExpanders } = prevState;
+      const prevExpanderOpen = prevOpenExpanders.includes(index);
+      const openExpanders = prevExpanderOpen
+        ? prevOpenExpanders.filter((value) => value !== index)
+        : Array.from(new Set([...prevOpenExpanders, index]));
+      return {
+        openExpanders,
+      };
+    });
+  };
+
+  isEvenOneExpanderClose = () => {
```

Comment:
isEvenOneExpanderClose*d* or allExpandersOpen or allExpandersClosed?

Reply:
This is now changed to `isEveryExpanderOpened`.